src/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,45 @@
+import { Component, Host, h, Prop, Watch, State } from "@stencil/core";
+
+@Component({
+  tag: "radio-group",
+  styleUrl: "radio-group.css",
+  shadow: true,
+})
+export class RadioGroup {
+  @Prop({ reflect: true }) list: Array<string>;
+  @State() selectedIndex: number;
+  @State() selectedItem: string;
+
+  handleChange = (e: any) => {
+    this.selectedIndex = parseInt(e.target.value);
+    this.selectedItem = e.target.name;
+  };
+
+  render() {
+    return (
+      <Host>
+        <slot>
+          <form>
+            <div>
+              {this.list.map((item, i) => {
+                return (
+                  <div>
+                    <input
+                      type="radio"
+                      name={item}
+                      value={i}
+                      checked={this.selectedIndex === i}
+                      onChange={this.handleChange}
+                    />
+                    <label>{item}</label>
```

Comment:
I like the use of the `<label>` tag here. However, should the input be inside of this tag or should the label have a `for` attribute for accessibility?

Reply:
I agree that`<label>` should have `for` attribute in-terms of accessibility in this use-case. Will make the change.

On the other hand ,we can also place input inside the `<label>` which make JSX code more readable and we don't need to add `for` attribute. With that approach any CSS styling applied to `label` tag will be implied to the input as well.

A small caveat is If we place the `<Input>` tag inside the `<label>` , the mouse click events will be fired if the user clicks on any open space inside the `<label>`.

Reply:
Yeah I think placing inside the label makes the most sense and easiest to implement.